packages/truffle-core/lib/command.js

```diff
@@ -100,9 +100,39 @@ class Command {
       }
     });
 
-    const newOptions = Object.assign({}, clone, argv);
-
+    //Check unsupported command line flag according to the option list in help
     try {
+      const inputOptions = [];
+      inputStrings.map(i => {
+        if (i.startsWith("--")) {
+          inputOptions.push(i);
+        }
+      });
+
+      const validOptions = [];
+      result.command.help.options.map(item => {
+        let opt = item.option.split(" ")[0];
+        if (opt.startsWith("--")) {
+          validOptions.push(opt);
+        }
+      });
+
+      let invalidOptions = inputOptions.filter(
+        opt => !validOptions.includes(opt)
+      );
+
+      if (invalidOptions.length > 0) {
+        if (options.logger) {
+          const log = options.logger.log || options.logger.debug;
+          log(
+            "warning: possilble unsupported (undocumented in help) command line option: " +
```

Comment:
Typo here, "possible unsupported". Also, for consistent formatting with other logging in Truffle, could we make this message something like "> Warning: possible...."?